mindsdb/integrations/handlers/confluence_handler/confluence_table.py

```diff
@@ -16,119 +16,98 @@ class ConfluencePagesTable(APITable):
 
-        if "space" in pages_kwargs:
-            confluence_pages_df = confluence_pages_df[
-                confluence_pages_df.space == pages_kwargs["space"]
-            ]
+        
+        confluence_pages_df.columns = confluence_pages_df.columns.str.replace("body.storage.value", "body")
+        confluence_pages_df['space'] = confluence_pages_df['space.key']
+        
+        available_columns = set(confluence_pages_df.columns).intersection(set(self.get_columns()))
+        confluence_pages_df = confluence_pages_df[list(available_columns)]
 
         selected_columns = []
         for target in query.targets:
             if isinstance(target, ast.Star):
-                selected_columns = self.get_columns()
+                selected_columns = list(available_columns)
                 break
             elif isinstance(target, ast.Identifier):
-                selected_columns.append(target.parts[-1])
+                col = target.parts[-1]
+                if col in available_columns:
+                    selected_columns.append(col)
+                else:
+                    raise ValueError(f"Unknown column: {col}")
             else:
                 raise ValueError(f"Unknown query target {type(target)}")
 
-        if len(confluence_pages_df) == 0:
-            confluence_pages_df = pd.DataFrame([], columns=selected_columns)
-        else:
-            confluence_pages_df.columns = self.get_columns()
-            for col in set(confluence_pages_df.columns).difference(
-                set(selected_columns)
-            ):
-                confluence_pages_df = confluence_pages_df.drop(col, axis=1)
+        confluence_pages_df = confluence_pages_df[selected_columns]
 
-            if len(order_by_conditions.get("columns", [])) > 0:
+        if query.order_by and len(query.order_by) > 0:
+            sort_columns = []
+            sort_ascending = []
+            for an_order in query.order_by:
+                if isinstance(an_order.field, ast.Identifier):
+                    column = an_order.field.parts[-1]
+                    if column in selected_columns:
+                        sort_columns.append(column)
+                        sort_ascending.append(an_order.direction == "ASC")
+                    else:
+                        raise ValueError(f"Order by unknown column {column}")
+                else:
+                    raise ValueError(f"Unsupported order by clause: {an_order}")
+            
+            if sort_columns:
                 confluence_pages_df = confluence_pages_df.sort_values(
-                    by=order_by_conditions["columns"],
-                    ascending=order_by_conditions["ascending"],
+                    by=sort_columns,
+                    ascending=sort_ascending
                 )
 
-        return confluence_pages_df
+        if query.limit:
+            confluence_pages_df = confluence_pages_df.head(query.limit.value)
```

Comment:
There is a limit being applied when calling the Confluence API. Is there a need to apply this again?

Reply:
@MinuraPunchihewa  thank you for reviewing the code i will fix this